RobotNavigation/Maze.cs

```diff
@@ -20,9 +21,13 @@ public enum Cell
        
         private Cell[,] _maze;
 
+        private ArrayList _posArray;
+        private Position _pos;
```

Comment:
I don't think the current position should be stored in the Maze object.  It should be stored inside the MazeSearch object instead.  The robots current position is not a property of a Maze, it's a property of a MazeSearch run.